spec.emu

```diff
@@ -142,14 +170,18 @@ contributors: Jordan Harband
 			1. Else if _type_ is `"Reject"`, let _handlerResult_ be Completion{[[Type]]: ~throw~, [[Value]]: _argument_, [[Target]]: ~empty~}.
 			1. Else if _type_ is `"Finally"`, let _handlerResult_ be NormalCompletion(_argument_).
 			1. Assert: _handlerResult_ is defined.
-		1. Else if _type_ is `"Finally"`, let _handlerResult_ be Call(_handler_, *undefined*, &laquo; &raquo;).
+		1. Else if _type_ is `"Finally"`, then
+			1. let _handlerResult_ be Call(_handler_, *undefined*, &laquo; &raquo;).
+			1. If _handlerResult_ is not an abrupt completion, then
+				1. Let _handlerPromise_ be ? PromiseResolve(%Promise%, _handlerResult_.[[Value]]).
+				1. Let _argumentThunk_ be equivalent to a function that returns _argument_.
```

Comment:
This text is identical to text that used to be in the spec prior to https://github.com/tc39/ecma262/pull/584 so I'm pretty sure it's OK. Not sure what you mean by "layering violation" tho?